assets/js/modules/analytics/hooks/useExistingTagEffect.js

```diff
@@ -28,97 +28,20 @@ import { useEffect } from '@wordpress/element';
  */
 import Data from 'googlesitekit-data';
 import { MODULES_ANALYTICS } from '../datastore/constants';
-import { CORE_MODULES } from '../../../googlesitekit/modules/datastore/constants';
-import { MODULES_TAGMANAGER } from '../../tagmanager/datastore/constants';
 const { useSelect, useDispatch } = Data;
 
 export default function useExistingTagEffect() {
-	const { setAccountID, selectProperty, setUseSnippet } = useDispatch(
-		MODULES_ANALYTICS
-	);
-	const gtmModuleActive = useSelect( ( select ) =>
-		select( CORE_MODULES ).isModuleActive( 'tagmanager' )
-	);
-
-	const {
-		existingTag,
-		existingTagAccountID,
-		existingTagPermission,
-		gtmAnalyticsPropertyID,
-		gtmAnalyticsAccountID,
-		gtmAnalyticsPermission,
-	} = useSelect( ( select ) => {
-		const data = {
-			existingTag: select( MODULES_ANALYTICS ).getExistingTag(),
-			existingTagPermission: false,
-			existingTagAccountID: '',
-			gtmAnalyticsPropertyID: '',
-			gtmAnalyticsAccountID: '',
-			gtmAnalyticsPermission: false,
-		};
+	const { setUseSnippet } = useDispatch( MODULES_ANALYTICS );
 
-		if ( data.existingTag ) {
-			// Just check existing tag permissions, if it is available and ignore tag manager settings.
-			const { permission = false, accountID = '' } =
-				select( MODULES_ANALYTICS ).getTagPermission(
-					data.existingTag
-				) || {};
-			if ( permission ) {
-				data.existingTagPermission = permission;
-				data.existingTagAccountID = accountID;
-			}
-		} else {
-			// There is no existing tag, so we need to try to get a property ID from GTM.
-			data.gtmAnalyticsPropertyID = select(
-				MODULES_TAGMANAGER
-			).getSingleAnalyticsPropertyID();
-			if ( data.gtmAnalyticsPropertyID ) {
-				const { permission = false, accountID = '' } =
-					select( MODULES_ANALYTICS ).getTagPermission(
-						data.gtmAnalyticsPropertyID
-					) || {};
-				if ( permission ) {
-					data.gtmAnalyticsAccountID = accountID;
-					data.gtmAnalyticsPermission = permission;
-				}
-			}
-		}
-
-		return data;
-	} );
+	const hasExistingTag = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).hasExistingTag()
+	);
 
 	useEffect( () => {
-		if ( existingTag ) {
-			// Disable the plugin snippet
+		if ( hasExistingTag ) {
```

Comment:
Reading this point in the IB, it seems to suggest that `useExistingTagEffect` should disable the snippet if there is an existing tag _and_ the ID is the same as the currently configured property ID. However, I can see that you have updated `UseSnippetSwitch` to contain this logic. This seems like it's probably the right thing to do, and also the IB is a little ambiguous in its phrasing. So, I just want to highlight this and make sure this is the right way to go about things here.

> Remove all use of getTagPermission from useExistingTagEffect and drastically simplify it so that all that it does going forward is disable the plugin snippet if the existing tag is the same as the currently configured property ID

Reply:
Yeah, I thought this was the best place to put it but I did notice that. Sorry I forgot to mention it here 👍🏻

Reply:
Makes sense, and no problem!

Reply:
I'd like to follow up on this thread, as @techanvil mentioned above, the `useSnippet` setting should indeed only be set to `false` if the existing tag is also the same as the currently set property ID. So it looks to me that this hook here needs that update still.

Regarding doing that in `UseSnippetSwitch`, that seems problematic to me since that component isn't necessarily used, for example we only display it in Analytics settings right now, but not in the Analytics setup. The logic to conditionally hide the snippet should apply in either case though.